### PR TITLE
 feat(test): Include Nuodb application component kill via pumba 

### PR DESCRIPTION
**Pull request by dargasudarshan**
**What this PR does / why we need it**:
- Include litmus book to perform Nuodb application component kill via pumba
- Verify the application components and ycsb load status post kill via liveness.
 
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by dargasudarshan** [comment]
Ansible log snippet:
```
TASK [Record the app_container] ************************************************
task path: /chaoslib/pumba/pod_failure_by_sigkill.yaml:43
2019-03-06T11:38:38.259402 (delta: 1.490884)         elapsed: 21.195122 *******
ok: [127.0.0.1] => {"ansible_facts": {"app_container": "sm"}, "changed": false}

TASK [Record the pumba pod on app node] ****************************************
task path: /chaoslib/pumba/pod_failure_by_sigkill.yaml:49
2019-03-06T11:38:38.458018 (delta: 0.198544)         elapsed: 21.393738 *******
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get pod -l app=pumba -o wide -n nuodbns | grep gke-e2e-cstor-dev-1-default-pool-224a4771-3j5j | awk '{print $1}'", "delta": "0:00:00.734980", "end": "2019-03-06 11:38:39.612859", "rc": 0, "start": "2019-03-06 11:38:38.877879", "stderr": "", "stderr_lines": [], "stdout": "pumba-vll5h", "stdout_lines": ["pumba-vll5h"]}

TASK [Record restartCount] *****************************************************
task path: /chaoslib/pumba/pod_failure_by_sigkill.yaml:58
2019-03-06T11:38:39.682814 (delta: 1.224746)         elapsed: 22.618534 *******
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get pod sm-0 -n nuodbns -o=jsonpath='{.status.containerStatuses[?(@.name==\"sm\")].restartCount}'", "delta": "0:00:00.762045", "end": "2019-03-06 11:38:40.713698", "rc": 0, "start": "2019-03-06 11:38:39.951653", "stderr": "", "stderr_lines": [], "stdout": "0", "stdout_lines": ["0"]}

TASK [Force kill the application pod using pumba] ******************************
task path: /chaoslib/pumba/pod_failure_by_sigkill.yaml:66
2019-03-06T11:38:40.791595 (delta: 1.108714)         elapsed: 23.727315 *******
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl exec pumba-vll5h -n nuodbns -- pumba kill --signal SIGKILL re2:k8s_sm_;", "delta": "0:00:01.676878", "end": "2019-03-06 11:38:42.677237", "rc": 0, "start": "2019-03-06 11:38:41.000359", "stderr": "time=\"2019-03-06T11:38:41Z\" level=info msg=\"Kill containers\" \ntime=\"2019-03-06T11:38:42Z\" level=info msg=\"Killing /k8s_sm_sm-0_nuodbns_b2382f9e-4003-11e9-9fa1-42010a800134_0 (00d3d134e3d4d6535899ff6ac593dc1ffb4490a2588a5c386f8561a32393270c) with signal SIGKILL\" ", "stderr_lines": ["time=\"2019-03-06T11:38:41Z\" level=info msg=\"Kill containers\" ", "time=\"2019-03-06T11:38:42Z\" level=info msg=\"Killing /k8s_sm_sm-0_nuodbns_b2382f9e-4003-11e9-9fa1-42010a800134_0 (00d3d134e3d4d6535899ff6ac593dc1ffb4490a2588a5c386f8561a32393270c) with signal SIGKILL\" "], "stdout": "", "stdout_lines": []}
TASK [obtain the number of replicas.] ******************************************
task path: /common/utils/check_deployment_status.yml:22
2019-03-06T11:38:47.427459 (delta: 0.088805)         elapsed: 30.363179 *******
changed: [127.0.0.1] => {"attempts": 1, "changed": true, "cmd": "kubectl get statefulset -n nuodbns -l nodetype=sm -o custom-columns=:spec.replicas", "delta": "0:00:00.625669", "end": "2019-03-06 11:38:48.352447", "rc": 0, "start": "2019-03-06 11:38:47.726778", "stderr": "", "stderr_lines": [], "stdout": "\n1", "stdout_lines": ["", "1"]}

TASK [Obtain the ready replica count and compare with the replica count.] ******
task path: /common/utils/check_deployment_status.yml:29
2019-03-06T11:38:48.489495 (delta: 1.061985)         elapsed: 31.425215 *******
changed: [127.0.0.1] => {"attempts": 1, "changed": true, "cmd": "kubectl get statefulset -n nuodbns -l nodetype=sm -o custom-columns=:..readyReplicas", "delta": "0:00:00.639695", "end": "2019-03-06 11:38:49.518601", "rc": 0, "start": "2019-03-06 11:38:48.878906", "stderr": "", "stderr_lines": [], "stdout": "\n1", "stdout_lines": ["", "1"]}

```

**Comment by dargasudarshan** [comment]
Post Chaos:

```
sudarshan_darga@cloudshell:~/pr/litmus/apps/nuodb (strong-eon-153112)$ kubectl get pods -n nuodbns
NAME                  READY     STATUS    RESTARTS   AGE
admin-0               1/1       Running   0          10m
nuodb-insights        2/2       Running   0          10m
sm-0                  1/1       Running   1          10m
te-79bbd6fc67-rjr68   1/1       Running   1          10m
ycsb-load-lm57x       1/1       Running   0          9m
```

File: apps/nuodb/chaos/app_pod_failure/test.yml [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+            ## APPLICATION FAULT INJECTION
+
+            - include_tasks: /chaoslib/pumba/pod_failure_by_sigkill.yaml
+              vars:
+                action: "killapp"
+                app_pod: "{{ app_pod_name.stdout }}"
+                namespace: "{{ app_ns }}"
+                label: "{{ app_label }}"
+
+          when: lookup('env','DEPLOY_TYPE') == 'statefulset'
```
Comment (gprasath) [inline apps/nuodb/chaos/app_pod_failure/test.yml]: @dargasudarshan Can you attempt this for deployment also, to make this litmusbook generic for all the nuodb components.(Both statefulset and deployment).

Comment (dargasudarshan) [reply apps/nuodb/chaos/app_pod_failure/test.yml]: Thank you @gprasath Will do that.

**Pull request merged**